compiler/luci-interpreter/src/kernels/StridedSlice.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/StridedSlice.h"
+
+#include "kernels/Utils.h"
+
+#include <tensorflow/lite/kernels/internal/reference/reference_ops.h>
+
+#include <stdexcept>
+
+namespace luci_interpreter
+{
+
+namespace kernels
+{
+
+const int kMaxDim = 4;
+
+inline int32_t positiveRemainder(int32_t dividend, int32_t divisor)
+{
+  return (divisor + (dividend % divisor)) % divisor;
+}
+
+inline int32_t clampedIndex(int32_t index, int dim, bool pos_stride)
+{
+  return pos_stride
+             ? (index >= dim ? dim : positiveRemainder(std::min(std::max(index, -dim), dim), dim))
+             : (index < -dim ? -1
+                             : positiveRemainder(std::min(std::max(index, -dim), dim - 1), dim));
+}
+
+StridedSlice::StridedSlice(const Tensor *input, const Tensor *begin, const Tensor *end,
+                           const Tensor *strides, Tensor *output, const StridedSliceParams &params)
+    : KernelWithParams<StridedSliceParams>(params), _input(input), _begin(begin), _end(end),
+      _strides(strides), _output(output)
+{
+}
+
+void StridedSlice::configure()
+{
+  assert(_begin->shape().num_dims() == 1);
+  assert(_end->shape().num_dims() == 1);
+  assert(_strides->shape().num_dims() == 1);
+  assert(_input->element_type() == _output->element_type());
+  assert(_begin->element_type() == DataType::S32);
+  assert(_end->element_type() == DataType::S32);
+  assert(_strides->element_type() == DataType::S32);
+  assert(_input->shape().num_dims() <= kMaxDim);
+  assert(params().ellipsis_mask == 0);
+  assert(params().new_axis_mask == 0);
+  if (_input->element_type() == DataType::U8)
+  {
+    assert(_input->scale() == _output->scale());
+    assert(_input->zero_point() == _output->zero_point());
+  }
+  std::vector<int32_t> output_shape_vector;
+  for (int i = 0; i < _input->shape().num_dims(); i++)
+  {
+    int _idx = _input->shape().num_dims() - i - 1;
+    int32_t _stride_value = getTensorData<int32_t>(_strides)[_idx];
+    assert(_stride_value != 0);
+    bool _pos_stride = _stride_value > 0;
+    int _dim = _input->shape().dim(_idx);
```

Comment:
We use underscore for member variables only.

Reply:
Okay, I will fix it!